### ISSUE TITLE
Mounted volumes flake with "No such file or directory"

### ISSUE THREAD

**Issue by fkorotkov** [cirruslabs/tart#567]
There is a bug in `virtiofs` integration that is reported as FB12594177 (please bring attention to it if you have contacts at Apple) where "No such file or directory" can be reported for a file on the mounted volume even thought these files are there.

Here are steps to reproduce with Tart:

```bash
git clone https://github.com/devMEremenko/XcodeBenchmark
tart clone ghcr.io/cirruslabs/macos-ventura-xcode:latest ventura-xcode
ventura-xcode tart run --dir=$PWD/XcodeBenchmark ventura-xcode 
```

Then inside the VM you can go to "/Volumes/My Shared Files" and run "sh benchmark.sh" to simulate the load.

**Comment by jlsalmon** [comment]
@fkorotkov Would you be open to the suggestion of adding FB12594177 to OpenRadar? http://openradar.appspot.com

**Comment by fkorotkov** [comment]
@jlsalmon thanks for noting OpenRadar. [Just duplicated it there](http://openradar.appspot.com/radar?id=5577789054910464).

On a side note there are bunch of reports of flakes and other bugs in `virtiofs` on macOS in [Docker for Mac issues](https://github.com/docker/for-mac/issues?q=is%3Aissue+is%3Aopen+virtiofs). Seems it's a more wide spread issue then we initially though. 😪

**Comment by sanieldalib** [comment]
Planning on testing out with Sonoma to see if this is still an issue? Has anyone tried it out yet?

**Comment by jlsalmon** [comment]
I did try a Sonoma host with a Ventura guest, and anecdotally experienced _fewer_ issues with my specific workload, but non-zero, and the benchmark still failed. Have not tried a Sonoma/Sonoma pair yet.

**Comment by fkorotkov** [comment]
Same here. Tested on some Betas for both host and guest back in July and still could reproduce. 😓 Haven't tried released version though!

**Comment by sanieldalib** [comment]
Is this still an issue through the latest 14.4.1 macOS release?

**Comment by sanieldalib** [comment]
@fkorotkov Thanks mate.

> As we understand this feature of Virtualization.Framework was envisioned for "sharing a few files from Desktop" and is not ready for heavy I/O.

Is this based on a response from Apple on your Feedback Ticket? If so, does that mean that we should not consider directory mounting as a viable option?

**Comment by fkorotkov** [comment]
Yeah, somewhat along the lines of my Feedback Ticket. Have no information if it will be improved in the next macOS version. I wouldn't put hopes on it.

**Comment by guiccbr** [comment]
Any news about this one or any workaround to be able to share a directory of my host with the vm?

**Comment by djtarazona** [comment]
> Make sure to create static size image if performance is important.

@bspinner Can you elaborate on this? I'm not familiar with the intricacies of disk images and best practices here. I've been treating sparse images via:
```
hdiutil create -size 100g -type SPARSE
```

Should I use UDIF for best performance?

```
hdiutil create -size 100g -type UDIF
```

Thanks!

**Comment by prohtex** [comment]
> Yeah, somewhat along the lines of my Feedback Ticket. Have no information if it will be improved in the next macOS version. I wouldn't put hopes on it.

Is this issue present in Sonoma guests as well? There was some talk of Sonoma Host above but I'm wondering if it is specific to Ventura guest.

I'm planning to host a high performance webserver with a few million tiny files inside a VM and I'll avoid virtiofs if it is buggy,

**Comment by marc-48k** [comment]
Yep @prohtex the also occurs on Sonoma. Unfortunately, it still persists on Sequoia too (15.2) I confirmed this last week.
